packages/appium/lib/cli/setup-command.js

```diff
@@ -0,0 +1,233 @@
+import log from '../logger';
+
+/**
+ * Subcommands of preset for setup
+ */
+export const SUBCOMMAND_MOBILE = 'mobile';
+export const SUBCOMMAND_DESKTOP = 'desktop';
+export const SUBCOMMAND_BROWSER = 'browser';
+
+/**
+ * A subcommand to remove APPIUM_HOME
+ */
+export const SUBCOMMAND_RESET = 'reset';
+
+/**
+ * Pairs of preset subcommand and driver candidates.
+ * Driver names listed in KNOWN_DRIVERS to install by default
+ */
+const PRESET_PAIRS = Object.freeze(
+  /** @type {const} */ ({
+    mobile: _.keys(MOBILE_DRIVERS),
+    desktop: _.keys(DESKTOP_DRIVERS),
+    browser: _.keys(DESKTOP_BROWSERS)
+  }),
+);
+const DRIVERS_ONLY_MACOS = ['xcuitest', 'safari', 'mac2'];
+
+/**
+ * Plugin names listed in KNOWN_PLUGINS to install by default.
+ */
+export const DEFAULT_PLUGINS = ['images'];
+
+/**
+ * Return a list of drivers available for current host platform.
+ * @param {import('appium/types').CliCommandSetupSubcommand} subcmd
+ * @returns {Array<string>}
+ */
+export function getPresetDrivers(subcmd) {
+  return _.filter(PRESET_PAIRS[subcmd], (driver) =>
+    (!system.isMac() && _.includes(DRIVERS_ONLY_MACOS, driver))
+    ? null
+    : driver
+  );
+}
+
+/**
+ * Return desktop platform name for setup command description.
+ * @returns {string}
+ */
+export function hostPlatformName() {
+  if (system.isMac()) {
+    return 'macOS';
+  } else if (system.isWindows()) {
+    return 'Windows';
+  }
+  return 'Linux';
+}
+
+/**
+ * Run 'setup' command to install drivers/plugins into the given appium home.
+ * @template {import('appium/types').CliCommandSetup} SetupCmd
+ * @param {import('appium/types').Args<SetupCmd>} preConfigArgs
+ * @param {string} appiumHome
+ * @param {DriverConfig} driverConfig
+ * @param {PluginConfig} pluginConfig
+ * @returns {Promise<void>}
+ */
+export async function runSetupCommand(appiumHome, preConfigArgs, driverConfig, pluginConfig) {
+  switch (preConfigArgs.setupCommand) {
+    case SUBCOMMAND_RESET:
+      await resetAppiumHome(appiumHome, driverConfig, pluginConfig);
+      break;
+    case SUBCOMMAND_DESKTOP:
+      await setupDesktopAppDrivers(driverConfig);
+      await setupDefaultPlugins(pluginConfig);
+      break;
+    case SUBCOMMAND_BROWSER:
+      await setupBrowserDrivers(driverConfig);
+      await setupDefaultPlugins(pluginConfig);
+      break;
+    default:
+      await setupMobileDrivers(driverConfig);
+      await setupDefaultPlugins(pluginConfig);
+      break;
+  }
+};
+
+/**
+ * Uninstall installed drivers and plugins in APPIUM_HOME. Then remove the entire Appium home.
+ * @param {string} appiumHome
+ * @param {DriverConfig} driverConfig
+ * @param {PluginConfig} pluginConfig
+ * @returns {Promise<void>}
+ */
+async function resetAppiumHome(appiumHome, driverConfig, pluginConfig) {
+  log.info(`Uninstalling drivers/plugins in APPIUM_HOME[${appiumHome}]`);
+  await uninstallDrivers(driverConfig);
+  await uninstallPlugin(pluginConfig);
+  await fs.rimraf(appiumHome);
```

Comment:
I still think we should not remove the whole folder root, but only the config yaml itself. If a user provides existing folder to be the root then we could potentially delete all their files

Reply:
Perhaps can have reset all, and reset. All removes entire home and reset is uninstall and yml file. For removing entire home would help users to start from scratch in user support

Reply:
I can only agree to that if we explicitly show a confirmation message to a user about we want to remove the folder ... recursively and they confirm this action explicitly

Reply:
Let me remove reset/uninstall related methods from this pr later to reduce the PR scope. It sounds like it would be nice to decide about the reset behavior in an issue first. Ordinary my thought was removing everything as a cleanup to restart setup from scratch

Reply:
-> https://github.com/appium/appium/issues/20117